collect_app/src/main/java/org/odk/collect/android/widgets/utilities/InternalRecordingRequester.java

```diff
@@ -77,13 +80,16 @@ private void handleRecording(RecordingSession session) {
                         formSaveViewModel.replaceAnswerFile(formIndex.toString(), result.getOrNull().getAbsolutePath());
                         Collect.getInstance().getFormController().answerQuestion(formIndex, new StringData(result.getOrNull().getName()));
                         refreshListener.onScreenRefresh();
+                    } else if (session.getId() instanceof Set) {
+                        Set<TreeReference> treeReferences = (Set<TreeReference>) session.getId();
+                        for (TreeReference treeReference : treeReferences) {
+                            Collect.getInstance().getFormController().getFormDef().setValue(new StringData(result.getOrNull().getName()), treeReference, false);
+                        }
                     }
-                } catch (JavaRosaException e) {
-                    // ?
-                }
 
-                if (!session.getId().equals("background")) {
                     session.getFile().delete();
+                } catch (JavaRosaException e) {
+                    // ?
```

Comment:
Can we do a Timber.e log and statically reach out to `ToastUtils` to show a message like "Saving background audio failed?"

Reply:
Sure! I'll add this in.